### PR TITLE
Story11 12

### PR DESCRIPTION
**Pull request by Flashdom**

### CODE REVIEW

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/AuthModel.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/AuthModel.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/RegModel.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/RegModel.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/User.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/User.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/UserModel.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/UserModel.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/UserToSend.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/UserToSend.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/api/question_model/googleModel.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.finapp.gramfin.finapp.api.question_model;
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/api/question_model/googleModel.java]: Move to auth_model package

File: app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/AuthFragment.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+    @OnClick(R.id.authorizebutton)
+    public void authorize() {
+        String email = emailet.getText().toString();
+        String password = passwordet.getText().toString();
+        authPresenter.apiService().authme(authPresenter.getUser(email, password)).enqueue(new Callback<AuthModel>() {
+            @Override
+            public void onResponse(Call<AuthModel> call, Response<AuthModel> response) {
+                AuthModel authModel = response.body();
+                authPresenter.saveData(email, password, authModel.user.id, authModel.user.token);
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/AuthFragment.java]: crash (on authorizebutton pressed):
    java.lang.NullPointerException: Attempt to read from field 'com.finapp.gramfin.finapp.api.question_model.UserModel com.finapp.gramfin.finapp.api.question_model.AuthModel.user' on a null object reference
        at com.finapp.gramfin.finapp.feature.authorization_fragment.view.AuthFragment$1.onResponse(AuthFragment.java:79)

File: app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/RegFragment.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        password=root.findViewById(R.id.passwordet);
+        regBut=root.findViewById(R.id.regbutton);
+        regBut.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                regPresenter.apiservice().regme(regPresenter.getUser(email.getText().toString(), password.getText().toString())).enqueue(new Callback<RegModel>() {
+                    @Override
+                    public void onResponse(Call<RegModel> call, Response<RegModel> response) {
+                        RegModel regModel = response.body();
+                        regPresenter.saveData(regModel.user.email, password.getText().toString(), regModel.user.id, regModel.user.token);
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/RegFragment.java]: crash(on regbutton pressed):
    java.lang.NullPointerException: Attempt to read from field 'com.finapp.gramfin.finapp.api.question_model.UserModel com.finapp.gramfin.finapp.api.question_model.RegModel.user' on a null object reference
        at com.finapp.gramfin.finapp.feature.authorization_fragment.view.RegFragment$1$1.onResponse(RegFragment.java:62)

File: app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/MyApplication.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.finapp.gramfin.finapp.feature.authorization_fragment.view;
+
+import android.app.Application;
+
+
+
+
+public class MyApplication extends Application {
```
Comment (antropit) [inline app/src/main/java/com/finapp/gramfin/finapp/feature/authorization_fragment/view/MyApplication.java]: Имя не отражает назначение, нужно переименовать

**Comment by Helga-Pataki** [comment]
все ок

**Pull request merged**